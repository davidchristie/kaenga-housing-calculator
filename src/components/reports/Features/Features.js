import PropTypes from 'prop-types'
import React, { Component } from 'react'
import { ListGroup, Well } from 'react-bootstrap'

import Feature from './Feature'

class Features extends Component {
  render () {
    const { features } = this.props
    return (
      <Well>
        <h2>Shared Features</h2>
        <ListGroup>
          {features.map((feature, index) => {
            return <Feature key={index} {...feature} />
          })}
        </ListGroup>
      </Well>
    )
  }
}

Features.defaultProps = {
  features: []
}

Features.propTypes = {
  features: PropTypes.arrayOf(
    PropTypes.shape({
      text: PropTypes.string.isRequired
    })
  ).isRequired
}

export default Features
