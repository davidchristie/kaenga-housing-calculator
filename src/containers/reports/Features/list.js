import { getHobbies } from '../../../selectors/hobbies'
import { getHomeBuilderValues } from '../../../selectors/home-builder'
import { getSustainability } from '../../../selectors/sustainability'

export default [
  {
    condition: state => true,
    text: 'Onsite car share service'
  },
  {
    condition: state => true,
    text: 'Close to public transport'
  },
  {
    condition: state => getHomeBuilderValues(state).study,
    text: 'Shared guest studios'
  },
  {
    condition: state => getHobbies(state).sports,
    text: 'Plenty of storage for your sports equipment'
  },
  {
    condition: state => getHobbies(state).exercising,
    text: 'Onsite gym and yoga studio'
  },
  {
    condition: state => getHobbies(state).building,
    text: 'Communal workshop/makers space'
  },
  {
    condition: state => getHobbies(state).gardening,
    text: 'Roof top flower and vegie garden'
  },
  {
    condition: state => getHobbies(state).games,
    text: 'A communal games room'
  },
  {
    condition: state => getHobbies(state).reading,
    text: 'A communal study/reading room'
  },
  {
    condition: state => getHobbies(state).artMusic,
    text: 'A music room'
  },
  {
    condition: state => getHobbies(state).artMusic,
    text: 'An onsite artist studio'
  },
  {
    condition: state => getHobbies(state).socialising,
    text: 'Indoor and outdoor entertainment spaces with a high end kitchen and barbeque'
  },
  {
    condition: state => true,
    text: 'North facing, double glazing, solar water heating'
  },
  {
    condition: state => ['medium', 'high'].includes(getSustainability(state).level),
    text: 'Water Reticulation, etc.'
  },
  {
    condition: state => getSustainability(state).level === 'high',
    text: 'Photovoltaic panels, etc.'
  }
]
