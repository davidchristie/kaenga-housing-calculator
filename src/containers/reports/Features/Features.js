import { connect } from 'react-redux'

import Features from '../../../components/reports/Features'

const features = [
  {
    condition: state => true,
    text: 'Onsite car share service'
  }
]

export default connect(
  state => {
    return {
      features: features
        .filter(feature => feature.condition(state))
    }
  }
)(Features)
