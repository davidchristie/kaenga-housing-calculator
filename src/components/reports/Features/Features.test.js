import React from 'react'
import ReactTestUtils from 'react-addons-test-utils'

import Features from './Features'

const shallowRenderer = ReactTestUtils.createRenderer()

it('matches snapshot', () => {
  expect(shallowRenderer.render(<Features />)).toMatchSnapshot()
})
