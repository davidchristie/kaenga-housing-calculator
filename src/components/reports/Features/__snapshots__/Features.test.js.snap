exports[`test matches snapshot 1`] = `
<Well
  bsClass="well">
  <h2>
    Shared Features
  </h2>
  <ListGroup
    bsClass="list-group">
    
  </ListGroup>
</Well>
`;
